src/app/3d/qgsmeshlayer3drendererwidget.cpp

```diff
@@ -37,6 +37,7 @@ QgsMeshLayer3DRendererWidget::QgsMeshLayer3DRendererWidget( QgsMeshLayer *layer,
   mWidgetMesh = new QgsMesh3dSymbolWidget( layer, this );
   mWidgetMesh->configureForDataset();
   layout->addWidget( mWidgetMesh );
+  mWidgetMesh->setEnabled( false );
```

Comment:
thanks for reporting (again) this issue. Indeed, #34999 didn't fix the issue... 
Instead, I would add this line here at line 60:
https://github.com/DelazJ/QGIS/blob/e9f1f71451a6b90c6f872dec3583f44f8945f32b/src/app
The widget was not set disable if  3D renderer was not provided...
Thanks again.